implement function for replacing german umlauts

simbo opened the pull request:
At the moment, german umlauts in titles are replaced with dashes, which doesn't result in nicely readable urls. A more "wordpress-like umlaut handling" would be nice:
"ä" replaced width "ae"
"ö" replaced width "oe"
"ü" replaced width "ue"
"ß" replaced width "ss"

I already implemented that in my fork, including tests. Maybe you can merge that...

Commit:
implement function for replacing german umlauts

ianstormtaylor commented:
Good point. Would be nicer if this was implemented for all types of accented characters I think, in that we'd use a different module that the slugifying one we currently use. Would you be down to edit the PR to handle that?

simbo committed:
update dependencies

simbo committed:
add script property to package json for running tests

simbo committed:
remove umlauts handling

simbo committed:
use 'slug' instead of 'slug-component' for slugifying; accept slug options

simbo committed:
add section about slug options to readme

simbo commented:
Great idea. Found [slug](https://www.npmjs.com/package/slug) which does a really nice job. It replaces all kinds of special characters and symbols...

I removed umlauts replacement and replaced 'slug-component' with 'slug'. The plugin now accepts slug options. I also added a test for accepting slug options and updated the readme.

sigo commented:
Bump. Is this will be ever merged? :)

woodyrew commented:
Thanks for the PR.

Could you resolve the conflicts or create a new PR based off master?

simbo commented:
I kinda can't believe it took over 3 years to update this pr...

Whatever... I'm not working with metalsmith anymore and don't want to spent time to update this.

But as this is an easy-to-implement and imho must-have feature (that should probably have been there much earlier :D), i am sure you will find someone else to do this.

I can still recommend to use [slug](https://www.npmjs.com/package/slug) for this job.
